README.md

```diff
@@ -16,25 +16,133 @@ specified bucket in GCS.
 ### File name format
 
 The connector uses the following format for output files (blobs):
-`<prefix><topic>-<partition>-<start_offset>[.gz]`, where:
-- `<prefix>` is the optional prefix that can be used, for example, for
-  subdirectories in the bucket;
-- `<topic>` is the Kafka topic name;
-- `<partition>` is the topic's partition number;
-- `<start_offset>` is the Kafka offset of the first record in the file;
-- `[.gz]` suffix is added when compression is enabled.
-
-The part after the prefix can be controlled by `file.name.template`. It
-supports placeholders with variable names: `{{ variable_name }}`. Only
-certain sets of variables are valid, currently:
-- `topic`, `partition`, `start_offset`.
-
-If not specified, the default value is
+`<prefix><filename>`.
+
+`<prefix>` is the optional prefix that can be used, for example, for
+subdirectories in the bucket.
+
+`<filename>` is the file name. The connector has the configurable
+template for file names. It supports placeholders with variable names:
+`{{ variable_name }}`. Currently supported variables are:
+- `topic` - the Kafka topic;
+- `partition` - the Kafka partition;
+- `start_offset` - the Kafka offset of the first record in the file;
+- `key` - the Kafka key.
+
+Only the certain combinations of variables are allowed in the file name
+template (however, variables in a template can be in any order). Each
+combination determines the mode of record grouping the connector will
+use. Currently supported combinations of variables and the corresponding
+record grouping modes are:
+- `topic`, `partition`, `start_offset` - grouping by the topic and
+  partition;
+- `key` - grouping by the key.
```

Comment:
Shouldn't this at least optionally allow specifying topic as well? Not being able to split by topic seems very restricting.

Reply:
For sure, but I plan to make it separately.